plugins/xpay/xpay.c

```diff
@@ -1569,15 +1572,46 @@ static struct command_result *json_xpay(struct command *cmd,
 		payment->amount = payment->full_amount;
 	}
 
-	/* Default is 5sats, or 1%, whatever is greater */
-	if (!maxfee) {
-		if (!amount_msat_fee(&payment->maxfee, payment->amount, 0, 1000000 / 100))
+	/* true if we have already computed the fee budget */
+	bool maxfee_set = false;
+
+	/* set maxfee through cmd option "maxfee" */
+	if (maxfee) {
+		payment->maxfee = *maxfee;
+		maxfee_set = true;
+	}
+
+	/* set maxfee through cmd option "maxfeepercent" */
+	if (maxfeepercent_ppm) {
+		struct amount_msat maxfee2;
+		if (!amount_msat_fee(&maxfee2, payment->amount, 0,
+				     *maxfeepercent_ppm / 100))
+			return command_fail(
+			    cmd, JSONRPC2_INVALID_PARAMS,
+			    "Invalid maxfeepercent: fee overflows");
+		if (!maxfee_set)
+			payment->maxfee = maxfee2;
+		else
+			/* one design choice: if we have two bounding limits
+			 * satisfy both. */
+			payment->maxfee =
+			    amount_msat_min(payment->maxfee, maxfee2);
+		maxfee_set = true;
+	}
+
+	/* Default is maxfee is 5sats, or 1%, whatever is greater */
```

Comment:
Why not have the same behaviour as [`pay`](https://docs.corelightning.org/reference/lightning-pay) with 0.5%? Imho it's a reasonable value that will get the payment through